src/Contacts/src/ContactPhaseList.cpp

```diff
@@ -140,12 +140,12 @@ bool ContactPhaseList::setLists(const std::initializer_list<ContactList>& contac
     return true;
 }
 
-ContactPhaseList::const_iterator ContactPhaseList::getPresentPhase(double time) const
+ContactPhaseList::const_iterator ContactPhaseList::getPresentPhase(double time, double tolerance /*= 0*/) const
 {
     // With the reverse iterator we find the last phase such that the begin time is smaller that
     // time
-    auto presentReverse = std::find_if(rbegin(), rend(), [time](const ContactPhase& a) -> bool {
-        return a.beginTime <= time;
+    auto presentReverse = std::find_if(rbegin(), rend(), [time, tolerance](const ContactPhase& a) -> bool {
+        return a.beginTime < time || std::abs(a.beginTime - time) <= tolerance;
```

Comment:
This should be equivalent to 
```
a.beginTime <= time + tolerance
```

In this way, you might also use negative tolerances to have the opposite effects, i.e. ``a.beginTime`` << ``time``.

Reply:
Yes you are right, I applied what you suggested in  9c1a49364634ca8d52439d19e444ce13c30efb85